src/assets/asset_manager.rs

```diff
@@ -62,31 +67,35 @@ impl AssetManager {
             }
             if file_name.ends_with(".gltf") {
                 let current_index = self.materials.len() as u32;
-                let (mesh, materials) = Mesh::new(
-                    &device,
-                    format!("{}{}", full_file_path, file_name),
-                    current_index,
-                );
-                let mut index = current_index;
-                for material in materials {
-                    self.materials.insert(index, material);
-                    index += 1;
+                let gltf_data =
+                    GltfData::load(&device, format!("{}{}", full_file_path, file_name)).unwrap();
+                // TODO: Figure out what this needs to look like
+                let mesh = gltf_data.mesh;
+                for (handle,submeshes) in mesh.data {
+                    self.materials.insert(handle, None);
                 }
-                self.meshes.insert(file_name.to_string(), mesh);
+                self.meshes.entry(file_name.to_string()).or_insert( mesh);
                 info!("Loaded mesh: {}", file_name);
             }
-            if file_name.ends_with(".png")
-                || file_name.ends_with(".jpg")
-                || file_name.ends_with(".hdr")
-            {
-                let image = Image::new(
-                    &device,
-                    &mut init_encoder,
-                    format!("{}{}", full_file_path, file_name),
-                    file_name.to_string(),
-                );
-                self.images.insert(file_name.to_string(), image);
+            //TODO: Dont load pictures here
+            if file_name.ends_with(".png") || file_name.ends_with(".jpg") {
+                let image;
+                if file_name.to_lowercase().contains("_normal")
+                    || file_name.to_lowercase().contains("metallic")
+                {
+                    image =
+                        Image::new_normal(&device, &mut init_encoder, entry.path().into()).unwrap();
+                } else {
+                    image =
+                        Image::new_color(&device, &mut init_encoder, entry.path().into()).unwrap();
+                }
+                self.images.insert(image.name, image);
                 info!("Loaded image: {}", file_name);
+            } else if file_name.ends_with(".hdr") {
+                let image =
+                    Image::new_hdr(&device, &mut init_encoder, entry.path().into()).unwrap();
+                self.images.insert(image.name, image);
+                info!("Loaded skybox: {}", file_name);
```

Comment:
The image code looks much cleaner thanks for cleaning this up! 👍

Reply:
:)